### PR TITLE
(MAINT) Document `Get-DocumentLink` and `Get-Document` cmdlets

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary

Prior to this change, the `Get-DocumentLink` and `Get-Document` functions didn't have any reference documentation.

This change:

- Scaffolds the Markdown documentation from the module
- Fills out the reference documentation for `Get-DocumentLink`
- Fills out the reference documentation for `Get-Document`

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://github.com/michaeltlombardi/DocumentarianModules/blob/main/CONTRIBUTING.md
[style]: https://github.com/michaeltlombardi/DocumentarianModules/blob/main/CONTRIBUTING.md#Style

### CODE REVIEW

File: Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-Document.md [hunk 1]
```diff
@@ -1,72 +1,62 @@
----
+﻿---
 title: Get-Document
-summary: |
-  `Get-Document` ...
+summary: Get-Document...
 external help file: Documentarian-help.xml
 Module Name: Documentarian
-online version:
+online version: 0.0.1
```
Comment (sdwheeler) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-Document.md:6]: This is a URL for `Get-Help -Online`

Comment (sdwheeler) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-Document.md:6]: This is a URL for `Get-Help -Online`

File: Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-DocumentLink.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
+﻿---
+title: Get-DocumentLink
+summary: Retrieves links from a Markdown document with their metadata.
+external help file: Documentarian-help.xml
+Module Name: Documentarian
+online version: 0.0.1
```
Comment (sdwheeler) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-DocumentLink.md:6]: This is a URL for `Get-Help -Online`

File: Source/Modules/Documentarian/Documentation/reference/cmdlets/_index.md [hunk 3]
```diff
@@ -2,8 +2,27 @@
 title: Cmdlet Reference
 linkTitle: Cmdlets
 summary: Reference documentation for the PowerShell cmdlets in the **Documentarian** module.
-description: >-
-  Reference documentation for the PowerShell cmdlets in the **Documentarian** module.
+description: Reference documentation for the PowerShell cmdlets in the **Documentarian** module.
+Module Name: Documentarian
+Module Guid: 2422ec90-815c-4c1d-8ec1-4c9b082f2909
+Download Help Link: '' # unset
+Help Version: ''       # unset
```
Comment (sdwheeler) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/_index.md:9]: These are for updateable help

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: Source/Modules/Documentarian/Documentation/reference/cmdlets/_index.md [hunk 4]
```diff
@@ -2,8 +2,25 @@
...
+Module Name: Documentarian
+Module Guid: 2422ec90-815c-4c1d-8ec1-4c9b082f2909
+Download Help Link: https://microsoft.github.io/Documentarian/modules/documentarian/help/
+Help Version: 0.0.1
+Locale: en-US
 ---
 
-Not yet written.
+The **Documentarian** module includes commands for inspecting, editing, and maintaining Markdown
+documents.
```
Comment (michaeltlombardi) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/_index.md:16]: I updated these values, but I also removed the module name header. I think we'll need to have a bit of noodling about this for the export-to-maml problem, _but_ we should approach that holistically.

The alternative is to leave the `_index.md` file without this information and keep the `Documentarian.md` file but marked as draft so Hugo never builds it.

File: Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-Document.md [hunk 5]
```diff
@@ -1,52 +1,146 @@
...
 
-## Synopsis
+## SYNOPSIS
+Parses a document as Markdown and returns it as a usable object.
+
+## SYNTAX
+
+```
+Get-Document [[-Path] <String[]>] [<CommonParameters>]
+```
```
Comment (michaeltlombardi) [inline Source/Modules/Documentarian/Documentation/reference/cmdlets/Get-Document.md:19]:
`````suggestion
```pwsh-syntax
Get-Document
 [[-Path] <String[]>]
 [<CommonParameters>]
```
``````

**Review by sdwheeler — CHANGES_REQUESTED** [review]

**Review by sdwheeler — DISMISSED** [review]
LGTM

**Pull request merged**